src/int.rs

```diff
@@ -471,6 +471,73 @@ impl I256 {
     pub fn to_little_endian(&self, bytes: &mut [u8]) {
         self.0.to_little_endian(bytes)
     }
+
+    /// Create 10**n as this type.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the result overflows the type.
+    pub fn exp10(n: usize) -> Self {
+        U256::exp10(n).try_into().expect("overflow")
+    }
+
+    /// Raise self to the power of `exp`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the result overflows the type in debug mode.
+    pub fn pow(self, exp: u32) -> Self {
+        handle_overflow(self.overflowing_pow(exp))
+    }
+
+    /// Raises self to the power of `exp`.
+    ///
+    /// Returns a tuple of the exponentiation along with a bool indicating
+    /// whether an overflow happened.
+    pub fn overflowing_pow(self, exp: u32) -> (Self, bool) {
+        let sign = match self.signum64().pow(exp & 1) {
```

Comment:
😭 🤕 🤯

I would have found logic like `if self.is_negative && exp % 2 ==> Sign Negative, else positive` much simpler to understand. But I guess this is correct too.

Reply:
Sorry! You’re right, `%` makes more sense.